roles/rsyslog/roles/input_roles/basics/defaults/main.yml

```diff
@@ -29,45 +27,99 @@ __rsyslog_basics_rules:
   - '{{ __rsyslog_conf_system_input }}'
   - '{{ __rsyslog_conf_network_input }}'
 
-# Debops example configuration options
-# --------------------------------------------
-
-# __rsyslog_conf_local_modules
+# __rsyslog_conf_local_basics_modules
 #
 # List of ``rsyslogd`` modules that enable logs from the local system to be
 # received and parsed by the ``rsyslogd`` daemon.
 __rsyslog_conf_local_basics_modules:
 
-  - name: 'local-basics-modules'
+  - name: 'basics-modules'
     type: 'modules'
     sections:
 
-      - comment: 'Log messages sent to local UNIX socket'
+      - comment: 'Reads kernel messages (the same are read from journald)'
         options: |-
+          module(load="imklog" permitnonkernelfacility="on")
+        state: '{{ "present"
+                  if ("kernel-message" in rsyslog_capabilities)
+                  else "absent" }}'
+
+      - comment: 'System log messages'
+        options: |-
+          # Log messages sent to local UNIX socket
           module(load="imuxsock"    # provides support for local system logging (e.g. via logger command)
           {% if rsyslog_use_imuxsock %}
                  SysSock.Use="on")  # Turn on message reception via local log socket.
           {% else %}
                  SysSock.Use="off") # Turn off message reception via local log socket.
           {% endif %}
-          # local messages are retrieved through imjournal now.
-          module(load="imjournal"             # provides access to the systemd journal
-                 StateFile="{{ rsyslog_work_dir }}/imjournal.state") # File to store the position in the journal
+          # Read from journald
+          {% set done = "false" %}
+          {% for input in logging_inputs %}
+          {%   if done == "false" and input.type == "basics" %}
+          module(load="imjournal"
+                 StateFile="{{ rsyslog_work_dir }}/imjournal.state"
+                 RateLimit.Burst="{{ input.rsyslog_imjournal_ratelimit_burst | d(20000) }}"
+                 RateLimit.Interval="{{ input.rsyslog_imjournal_ratelimit_interval | d(600) }}"
+                 PersistStateInterval="{{ input.rsyslog_imjournal_persist_state_interval | d(10) }}")
+          {%     set done = "true" %}
```

Comment:
Does this actually work?  according to https://jinja.palletsprojects.com/en/2.11.x/templates/#assignments you aren't supposed to do this.  If you need to use the value of `done` later on in the code, use a namespace.  Otherwise, can you use or one of the other loop special variables, like `loop.first` or `loop.last`, listed in the table under https://jinja.palletsprojects.com/en/2.11.x/templates/#for?

Reply:
It may not be.  When I tested this, I thought it fixed my problem (and it looked it made the CI test pass...)  But in the later coding in other places, I found I had to use namespace.  Fixing this.  Thanks!